### PR TITLE
infra: migrate travis jobs to semaphore

### PR DESCRIPTION
**Pull request by nrmancuso**
Migrates most travis jobs to semaphore, travis has been failing for quite awhile now. We will keep "test and deploy" and snapshot deployment in travis on master branch until at least https://github.com/checkstyle/checkstyle/issues/12620

### CODE REVIEW

**Review by romani — APPROVED** [review]

File: .semaphore/semaphore.yml [hunk 1]
```diff
@@ -27,6 +27,33 @@ blocks:
...
+            - mvn -e --no-transfer-progress clean integration-test failsafe:verify "${ARGS}"
+
+        - name: sevntu
+          commands:
+            - mvn -e --no-transfer-progress compile antrun:run@ant-phase-verify-sevntu -Psevntu
+
+        - name: codecov
+          commands:
+            - export USE_MAVEN_REPO="true"
+            - ./.ci/travis.sh jacoco
```
Comment (rnveach) [inline .semaphore/semaphore.yml:42]: Same as before, script needs to change.

File: .semaphore/semaphore.yml [hunk 2]
```diff
@@ -27,6 +27,33 @@ blocks:
...
+
+        - name: codenarc analysis
+          commands:
+            - ./.ci/codenarc.sh
+
+        - name: no-exception-samples-ant
+          commands:
+            - ./.ci/no-exception-test.sh no-exception-samples-ant
```
Comment (rnveach) [inline .semaphore/semaphore.yml:50]: Should we just put these with other no-exceptions?

File: .semaphore/semaphore.yml (hunk 1 above)
Comment (rnveach) [reply .semaphore/semaphore.yml]: This was done now.

File: .semaphore/semaphore.yml (hunk 2 above)
Comment (nrmancuso) [reply .semaphore/semaphore.yml:49]: Done

Comment (nrmancuso) [reply .semaphore/semaphore.yml:49]: Wait, not done

File: .travis.yml [hunk 3]
```diff
@@ -1,94 +0,0 @@
...
-  # - ./.ci/travis.sh remove-custom-mvn
-  # - ./.ci/travis.sh remove-adoptium-jdk
-  - ./.ci/validation.sh git-diff
-  - ./.ci/travis.sh ci-temp-check
-  - ./.ci/travis.sh quarterly-cache-cleanup
-  - sleep 5s
-
-after_success:
-  - ./.ci/travis.sh run-command-after-success
-  - ./.ci/travis.sh deploy-snapshot
```
Comment (romani) [inline .travis.yml:93]: Who will do this ?

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .semaphore/semaphore.yml (hunk 2 above)
Comment (nrmancuso) [reply .semaphore/semaphore.yml]: Ok, done now

File: .travis.yml [hunk 4]
```diff
@@ -1,94 +0,0 @@
...
-
-script:
-  # manually set JAVA_HOME to overcome issue with travis ci noted at
-  # https://github.com/checkstyle/checkstyle/pull/11699#issue-1261272652
-  # - JAVA_HOME='/usr/lib/jvm/adoptopenjdk-11-hotspot-ppc64el'
-  - ./.ci/travis.sh init-m2-repo
-  - ./.ci/travis.sh run-command "$CMD"
-  # - ./.ci/travis.sh remove-custom-mvn
-  # - ./.ci/travis.sh remove-adoptium-jdk
-  - ./.ci/validation.sh git-diff
```
Comment (rnveach) [inline .travis.yml:86]: More just a question tagging onto @romani 's, should we be ensure all these other CIs run a git-diff check?

File: .travis.yml [hunk 5]
```diff
@@ -1,94 +0,0 @@
...
-  - ./.ci/travis.sh ci-temp-check
```
Comment (rnveach) [inline .travis.yml:87]: Does our other CI's do this for `.ci-temp` usage?

File: .travis.yml (hunk 3 above)
Comment (nrmancuso) [reply .travis.yml:93]: On it now

File: .travis.yml (hunk 4 above)
Comment (nrmancuso) [reply .travis.yml:86]: I will add this to all

File: .travis.yml (hunk 5 above)
Comment (nrmancuso) [reply .travis.yml:87]: Moving this elsewhere

File: .travis.yml (hunk 3 above)
Comment (nrmancuso) [reply .travis.yml]: Ok, travis seems to be more stable in master branch, let's leave only this task (`tests and deploy` and `deploy-snapshot`) in travis.

File: .travis.yml (hunk 4 above)
Comment (nrmancuso) [reply .travis.yml]: We have this in a bunch of CI:
```
➜  checkstyle git:(move-travis-tasks) ✗ ag -Q "./.ci/validation.sh git-diff" --hidden
.circleci/config.yml
53:          command: ./.ci/validation.sh git-diff

.semaphore/semaphore.yml
64:            - ./.ci/validation.sh git-diff
86:            - ./.ci/validation.sh git-diff
105:            - ./.ci/validation.sh git-diff

azure-pipelines.yml
167:      ./.ci/validation.sh git-diff

.travis.yml
54:  - ./.ci/validation.sh git-diff


```

And it is already in semaphore, so "done"

File: .travis.yml (hunk 5 above)
Comment (nrmancuso) [reply .travis.yml]: Added to other CI in separate commit

File: .travis.yml (hunk 3 above)
Comment (nrmancuso) [reply .travis.yml]: Issue is created at https://github.com/checkstyle/checkstyle/issues/12620

**Review by rnveach — APPROVED** [review]

**Review by romani — APPROVED** [review]
if CI pass  , ok to merge

**Pull request merged**